apps/api_web/lib/api_web/rate_limiter/rate_limiter_concurrent.ex

```diff
@@ -0,0 +1,167 @@
+defmodule ApiWeb.RateLimiter.RateLimiterConcurrent do
+  @moduledoc """
+  Rate limits a user's concurrent connections based on their API key or by their IP address if no
+  API key is provided. Split by static and event-stream requests.
+  """
+
+  use GenServer
+  require Logger
+  alias ApiWeb.RateLimiter.Memcache.Supervisor
+
+  @rate_limit_concurrent_config Application.compile_env!(:api_web, :rate_limiter_concurrent)
+  @uuid_key "ApiWeb.RateLimiter.RateLimiterConcurrent_uuid"
+  def start_link([]), do: GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+
+  def init(_) do
+    uuid = UUID.uuid1()
+    :persistent_term.put(@uuid_key, uuid)
+    {:ok, %{uuid: uuid}}
+  end
+
+  defp lookup(%ApiWeb.User{} = user, event_stream?) do
+    type = if event_stream?, do: "event_stream", else: "static"
```

Comment:
question: it doesn't look like `type` is ever used?

Reply:
Removed